src/main/java/ratpack/zipkin/internal/ZipkinHttpClientImpl.java

```diff
@@ -81,18 +75,16 @@ public void close() {
 
     @Override
     public Promise<ReceivedResponse> request(URI uri, Action<? super RequestSpec> action) {
-        return delegate.request(uri, tracedRequestAction(action))
-                .wiretap(receivedResponseResult -> responseInterceptor
-                        .handle(responseAdapterFactory.createAdapter(receivedResponseResult.getValue()))
-                );
+        final ExecuteWithSpan execution = new ExecuteWithSpan(tracer);
+        return delegate.request(uri, execution.execute(action))
```

Comment:
Key part is brave 4 spans are started independent of their creation exactly for this reason..

tracer.nextSpan(); for example, will create a span.. you can set that in scope independent of its timing, and setting it will allow others to read it (ex to add tags or read trace IDs).